### PR TITLE
add `gleam new` flags to control git init and .github content creation

### PR DESCRIPTION
**Pull request by bookshelfdave**
Addresses the cli feature request from #1875.

### CODE REVIEW

File: compiler-cli/src/new.rs [hunk 1]
```diff
@@ -72,14 +72,22 @@ impl Creator {
...
+        }
+        if self.options.git_init {
+            crate::fs::git_init(&self.root)?;
+        }
 
         match self.options.template {
             Template::Lib => {
-                self.gitignore()?;
-                self.github_ci()?;
+                if self.options.git_init {
```
Comment (bookshelfdave) [inline compiler-cli/src/new.rs:85]: I wasn't sure if this should be lumped in with `git init` functionality. What if the dev is using something other than git?

File: compiler-cli/src/new.rs [hunk 2]
```diff
@@ -72,14 +72,22 @@ impl Creator {
...
+                    self.gitignore()?;
```
Comment (bookshelfdave) [inline compiler-cli/src/new.rs:86]: I wasn't sure if this should be lumped in with `git init` functionality. What if the dev is using something other than git?

**Comment by inoas** [comment]
IMHO at least git should be opt-out not opt-in as it seems to be a sane default.

Also it feels like github files make no sense if git isn't there so these options depend on each other.

```shell
--skip=git # also sets no-github
--skip=github # just opt out of github files but not git
```

```shell
--no-git # also sets no-github
--no-github
``

```shell
--skeleton=github # default
--skeleton=gitlab
--skeleton=none
--ci=github
--ci=gitlab
--ci=none
``

**Review by lpil — COMMENTED** [review]
Wonderful! Let's do as @inoas says and default to rendering everything.

We can have a `--skip-git` which disables both, and `--skip-github` which disables the `.github` stuff.

File: CHANGELOG.md [hunk 3]
```diff
@@ -5,6 +5,10 @@
 - 4 digit integers are now always formatted without underscores.
 - Running `gleam new` will skip `git init` if the new project directory is
   already part of a git work tree.
+- The `gleam new` command gains the `--skip-git` flag to skip initialization
+  of both git and .github content.
```
Comment (inoas) [inline CHANGELOG.md:9]: The `gleam new` command gains the `--skip-git` flag to skip initialization of `.git/*`, `.gitignore` and `.github/*` files.

File: CHANGELOG.md [hunk 4]
```diff
@@ -5,6 +5,10 @@
...
+- The `gleam new` command gains the `--skip-github` flag to skip initialization
+  of .github content.
```
Comment (inoas) [inline CHANGELOG.md:11]: The `gleam new` command gains the `--skip-github` flag to skip creation of `.github/*`  files.

File: compiler-cli/src/new.rs (hunk 2 above)
Comment (inoas) [reply compiler-cli/src/new.rs:88]: If this skips .gitignore copy-over if skip-git is set, I think that's sane _imho_.

File: compiler-cli/src/main.rs [hunk 5]
```diff
@@ -235,6 +235,14 @@ pub struct NewOptions {
         default_value = "lib"
     )]
     pub template: new::Template,
+
+    /// Skip git initialization and creation of .github content
```
[the remark this answers is not in the document]
Comment (bookshelfdave) [reply compiler-cli/src/main.rs:239]: per https://github.com/gleam-lang/gleam/pull/1901#pullrequestreview-1223052086, `--skip-git` should also skip `.github/*`

[the remark this answers is not in the document]
Comment (inoas) [reply compiler-cli/src/main.rs]: sure but the comment just says github

**Comment by bookshelfdave** [comment]
changes pushed

**Review by lpil — APPROVED** [review]
Brilliant, thank you very much!

**Pull request merged**